### ISSUE TITLE
Version >=2.0.5 on Ubuntu 20.04 returns "version `GLIBC_2.XX' not found (required by alpaca)"

### ISSUE THREAD

**Issue by bigbrozer** [samuong/alpaca#130]
Running alpaca from `alpaca_v2.0.7_linux-amd64` on Ubuntu 20.04 LTS (glibc 2.31):

```sh
❯ alpaca 
alpaca: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by alpaca)
alpaca: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by alpaca)
```

I'm not sure where it does come from but this issue is starting with version >=2.0.5 (prior versions are running fine).

**Comment by samuong** [comment]
Hi @bigbrozer, thanks for reporting this. The v2.0.5 release was the first one to use Go 1.22 (instead of Go 1.17) to build the release binary, and I suspect this change causes it to link against a newer glibc. I think it was some of the updated dependencies that pulled in this new Go version, so it might not be possible to move back to an older Go toolchain.

Both [v2.0.4](https://github.com/samuong/alpaca/actions/runs/9337085051/job/25698522787#step:1:5) and [v2.0.5](https://github.com/samuong/alpaca/actions/runs/9641841981/job/26588363227#step:1:5) were built on Ubuntu 22.04.4 runners, and I'm wondering if going back down to Ubuntu 20.4 for this instead. I've given this a try (see https://github.com/samuong/alpaca/actions/runs/9768982323), are you able to download this pre-release binary and let me know if it doesn't work?

https://github.com/samuong/alpaca/actions/runs/9768982323/artifacts/1661674874

**Comment by bigbrozer** [comment]
Hello @samuong, thank you very much for this ! It is working like a charm. Additionally, I have performed some tests on Ubuntu 22.04, 24.04 and Archlinux (latest glibc) systems, and this is working perfectly :-)

**Comment by samuong** [comment]
That's great to hear, thanks for testing this! I've just released this as v2.0.8: https://github.com/samuong/alpaca/releases/tag/v2.0.8
